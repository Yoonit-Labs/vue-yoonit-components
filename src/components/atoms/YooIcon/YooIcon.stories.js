import YooAvatar from './YooAvatar.vue'
import PropsConfig from '@/config/Props.config'

export default {
  title: 'Yoonit/YooAvatar',
  component: YooAvatar,
  argTypes: {
    variation: {
      control: {
        type: 'select',
        options: PropsConfig.variation.options
      }
    },
    buttonSize: {
      control: {
        type: 'select',
        options: PropsConfig.buttonSize.options
      }
    },
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    iconPosition: {
      control: {
        type: 'select',
        options: PropsConfig.iconPosition.options
      }
    },
    iconSize: {
      control: {
        type: 'select',
        options: PropsConfig.iconSize.options
      }
    },
    iconStyle: {
      control: {
        type: 'select',
        options: PropsConfig.iconStyle.options
      }
    },
    animation: {
      control: {
        type: 'select',
        options: PropsConfig.animation.options
      }
    },
    iconCustomColor: {
      control: {
        type: 'text'
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { YooAvatar },
  template:
    '<yoo-avatar v-bind="$props"></yoo-avatar>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Avatar' }

export const Outline = Template.bind({})
Outline.args = { text: 'Yoo Avatar', variation: 'outline' }
