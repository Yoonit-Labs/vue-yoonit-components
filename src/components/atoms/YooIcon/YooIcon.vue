<template lang="pug">
  i(
    :class="['yoo-icon', takeIconStyle, takeIconName, ...takeModifier]"
  )
</template>

<script>

export default {
  name: 'YooIcon',
  props: {
    icon: {
      type: String,
      default: ''
    },
    iconStyle: {
      type: String,
      default: 'solid',
      validator: value =>
        [
          'solid',
          'regular'
        ]
          .includes(value)
    },
    event: {
      type: String,
      required: false,
      default: ''
    },
    size: {
      type: String,
      default: 'md',
      validator: value =>
        [
          'xxxs',
          'xxs',
          'xs',
          'sm',
          'md',
          'lg',
          'xl',
          'xxl',
          'xxxl',
          'display'
        ]
          .includes(value)
    },
    fill: {
      type: String,
      default: 'neutral',
      validator: value =>
        [
          'primary',
          'danger',
          'neutral',
          'light',
          'dark'
        ]
          .includes(value)
    }
  },
  data: () => ({}),
  created () {},
  mounted () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  components: {},
  computed: {
    takeIconName () {
      return `fa-${this.icon}`
    },
    takeIconStyle () {
      return this.iconStyle === 'solid'
        ? 'fas' // fontawesome solid
        : 'far' // fontawesome regular
    },
    takeModifier () {
      const block = 'yoo-icon'
      const classList = []

      if (this.size) {
        classList
          .push(
            `${block}__size--${this.size}`
          )
      }

      if (this.fill) {
        classList
          .push(
            `${block}__fill--${this.fill}`
          )
      }
      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooIcon.sass" lang="sass" scoped></style>
