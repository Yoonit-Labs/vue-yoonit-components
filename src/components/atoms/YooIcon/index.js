import YooIcon from './YooIcon.vue'
export default YooIcon
