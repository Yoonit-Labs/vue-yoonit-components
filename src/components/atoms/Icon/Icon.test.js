import { shallowMount } from '@vue/test-utils'
import YooIcon from '@/components/atoms/Icon/Icon.vue'
import PropsConfig from '@/components/atoms/Icon/Icon.config'

const classBlock = 'yoo-icon'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooIcon, {
    slots: { default: SlotText }
  })
}

describe('YooIcon Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-icon').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('icon', () => {
      it('Has a valid default value', () => {
        expect(YooIcon.props.icon.default).toBe('')
      })
      it('Includes icon class: .fa-cog', async () => {
        await wrapper.setProps({ icon: 'cog' })
        expect(wrapper.find('.fa-cog').exists()).toBe(true)
      })
    })
    describe('iconStyle', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.iconStyle.options.includes(YooIcon.props.iconStyle.default)).toBe(true)
      })
      PropsConfig.iconStyle.options.forEach(iconStyle => {
        it(`Includes iconStyle class: .${iconStyle === 'solid' ? 'fas' : 'far'}`, async () => {
          await wrapper.setProps({ icon: 'icon-value', iconStyle })
          expect(wrapper.find('.yoo-icon').classes().includes(`${iconStyle === 'solid' ? 'fas' : 'far'}`)).toBe(true)
        })
      })
    })
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooIcon.props.size.default)).toBe(true)
      })
      PropsConfig.size.options.forEach(size => {
        it(`Includes variation class: .${classBlock}__size--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}__size--${size}`).exists()).toBe(true)
        })
      })
    })
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooIcon.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Includes variation class: .${classBlock}__fill--${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}__fill--${fill}`).exists()).toBe(true)
        })
      })
    })
  })
})
