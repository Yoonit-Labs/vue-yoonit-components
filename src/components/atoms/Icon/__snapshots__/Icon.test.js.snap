// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooIcon Component Matches Snapshot 1`] = `<i class="yoo-icon fas fa- yoo-icon__size--lg yoo-icon__fill--neutral"></i>`;
